providers/oauth/oauth.localpageauth.js

```diff
@@ -76,6 +76,18 @@ LocalPageAuth.prototype.launchAuthFlow = function(authUrl, stateObj, interactive
   window.addEventListener("storage", listener, false);
   // Start 'er up
   window.open(authUrl);
+
+  if (interactive === false) {
+    setTimeout(function() {
+      if (this.listeners[stateObj.state]) {
+        // Listener has not been deleted, indicating oauth has completed.
+        window.removeEventListener(
+            "storage", this.listeners[stateObj.state], false);
+        delete this.listeners[stateObj.state];
+        continuation(undefined, 'Error launching auth flow');
+      }
+    }.bind(this), 5000);
```

Comment:
Would be good to move this hardcoded timeout to a constant at the top (e.g. var timeout = 5000).